src/main/java/sparksoniq/jsoniq/runtime/iterator/postfix/PredicateIterator.java

```diff
@@ -19,23 +19,22 @@
  */
 package sparksoniq.jsoniq.runtime.iterator.postfix;
 
-import sparksoniq.exceptions.InvalidSelectorException;
 import sparksoniq.jsoniq.item.IntegerItem;
 import sparksoniq.jsoniq.item.Item;
-import sparksoniq.jsoniq.runtime.iterator.primary.IntegerRuntimeIterator;
 import sparksoniq.jsoniq.runtime.metadata.IteratorMetadata;
 import sparksoniq.semantics.DynamicContext;
 import sparksoniq.exceptions.SparksoniqRuntimeException;
 import sparksoniq.exceptions.IteratorFlowException;
 import sparksoniq.jsoniq.runtime.iterator.LocalRuntimeIterator;
 import sparksoniq.jsoniq.runtime.iterator.RuntimeIterator;
-import sparksoniq.semantics.types.ItemType;
-import sparksoniq.semantics.types.ItemTypes;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class PredicateIterator extends LocalRuntimeIterator {
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.function.Function;
+
+public class PredicateIterator extends RuntimeIterator {
```

Comment:
I may not be up to date with all the changes but shouldn't this be a SparkRuntimeIterator?

Reply:
Or it can work both ways now actually. We should probably introduce a new base class for this case.

Reply:
> Or it can work both ways now actually. We should probably introduce a new base class for this case.

How about SparkOrLocalIterator?